webapp/src/monaco.tsx

```diff
@@ -795,19 +799,95 @@ export class Editor extends toolboxeditor.ToolboxEditor {
         })
     }
 
-    private insertSnippet(position: monaco.Position, insertText: string) {
+    private insertSnippet(cursorPos: monaco.Position, insertText: string, inline = false) {
         let currPos = this.editor.getPosition();
         let model = this.editor.getModel();
-        if (!position) // IE11 fails to locate the mouse
-            position = currPos;
+        if (!cursorPos) // IE11 fails to locate the mouse
+            cursorPos = currPos;
+        if (!cursorPos) {
+            return
+        }
+        let position = cursorPos.clone()
+
+        // determine insert mode
+        type InsertMode = "NewLineBefore" | "NewLineAfter" | "Inline"
+        // TODO: allow expressions to be inserted into the middle of the line
+        let insertMode: InsertMode;
+        if (inline)
+            insertMode = "Inline"
+        else if (cursorPos.column === 1)
+            insertMode = "NewLineBefore"
+        else
+            insertMode = "NewLineAfter"
+
+        // determine snippet insert column
+        if (insertMode === "NewLineAfter") {
+            // place snippet as if the cursor where at the end of the line
+            position.column = model.getLineMaxColumn(cursorPos.lineNumber)
+        } // else no change
+
+        // if not inline, put snippet on a new line
+        const cursorLineContent = model.getLineContent(cursorPos.lineNumber)
+        const cursorLineIsEmpty = !cursorLineContent
+        if (insertMode === "Inline") {
+            // TODO
```

Comment:
is there a note you can add for the todo here / is there still something that needs doing with shannons change in this branch?

Reply:
Actually there was nothing todo so I removed it